### ISSUE TITLE
aarch64 build

### ISSUE THREAD

**Issue by davemackintosh** [loadimpact/k6#842]
Hi there,

Started using this today and it's just fantastic, very well done all involved. Works amazingly on my laptop but my alternate development environment is my phone/termux and the apt repo reports no support for the aarch64 architecture and golang is broken on Termux.

Any chance of an aarch64 build for us on mobile?

Thanks again!

**Comment by na--** [comment]
Compiling k6 for aarch64 should be relatively easy and you should be able to do it from any other OS like this:
```bash
git clone https://github.com/loadimpact/k6.git
cd k6
GOARCH=arm64 GOOS=linux go build -o k6-aarch64
```

Though I can't guarantee that k6 will work properly on that architecture. There's no reason why it shouldn't, but at this point, to officially support it, we should also run the CI test suite on an ARM machine, which could be a bit trickier to set up...

**Comment by davemackintosh** [comment]
I'll cross compile it today on the laptop and test. Agree on the CI, most likely scenario would be to run qemu to emulate ARM. I can take a look tomorrow

**Comment by davemackintosh** [comment]
The build works as expected on my android device, I'll see if I can PR to the circleCI config an ARM test runner in the coming days.

**Comment by grantyoung** [comment]
Any update on this? With the switch to Arm on Macs I think that architecture will be becoming more popular quite soon and quite quickly.

**Comment by na--** [comment]
Yeah, it's probably high time we do this... The hard isn't releasing the ARM binaries, but running the tests on an ARM machine. We'd probably have to set up a self-hosted runner. Or just release without testing... :sweat_smile: 
 
Regarding Macs, I think proper M1 support is coming in Go 1.16. Assuming that comes out in the next week or two and no scary bugs pop up in the Go issue tracker, it's likely that we'll release k6 v0.31.0 with it in early March. If Go 1.16 gets delayed though, we're going to still use 1.15.x for k6 v0.31.0

**Comment by grantyoung** [comment]
Many thanks @na--. If it helps [AWS does have Arm instances now](https://aws.amazon.com/ec2/instance-types/a1/) :)

**Comment by na--** [comment]
Sorry, once again, we didn't have time for this... :disappointed: It's also very likely that we will release k6 v0.31.0 with Go 1.15.x anyway, but still... Moving this to k6 v0.32.0 with expected release date May 3rd. If anyone needs it before that, k6 is easy to build from scratch.

**Comment by 24HOURSMEDIA** [comment]
Waiting for an official release, I created a docker image for arm/linux and tested it on a Raspberry Pi 4 with Ubuntu server 20.04LTS / AR64 and it seems to run fine - 

https://github.com/24HOURSMEDIA/docker-k6-xarch
https://hub.docker.com/repository/docker/24hoursmedia/k6-xarch

**Comment by grantyoung** [comment]
Many thanks for this!
